xiblint/__main__.py

```diff
@@ -47,25 +51,40 @@ def main():
     #
     # Process paths
     #
-    success = True
+    errors = []
     for path in config.include_paths:
         for root, _, files in os.walk(path):
             for file_path in [os.path.join(root, file) for file in files]:
                 checkers = config.checkers(file_path)
-                success = process_file(file_path, checkers) and success
+                errors += process_file(file_path, checkers, args.reporter)
 
-    sys.exit(0 if success else 1)
+    print_errors(errors, args.reporter)
 
+    sys.exit(len(errors) == 0)
```

Comment:
That looks wrong. You want to exit with zero for success and non-zero for failure. You can also simply say:
```
sys.exit(1 if errors else 0)
```

Reply:
Oh, my mistake. Was so ashamed of this line that I quickly copied/pasted this one and wanted to forget about it 😓 Now tested the statement using the python interpreter and it works for an empty array as well 🤔 Didn't know that it worked that way as well ;-)